tree.go

```diff
@@ -443,9 +443,9 @@ func (n *InternalNode) GetCommitment() *bls.G1Point {
 	return n.commitment
 }
 
-func (n *InternalNode) GetCommitmentsAlongPath(key []byte) ([]*bls.G1Point, []*bls.Fr, []*bls.Fr, [][]bls.Fr) {
-	childIdx := Offset2Key(key, n.depth, n.treeConfig.width)
-	comms, zis, yis, fis := n.children[childIdx].GetCommitmentsAlongPath(key)
+func (n *InternalNode) getCommitmentsAlongPath(keys [][]byte) ([]*bls.G1Point, []*bls.Fr, []*bls.Fr, [][]bls.Fr) {
+	childIdx := Offset2Key(keys[0], n.depth, n.treeConfig.width)
+	comms, zis, yis, fis := n.children[childIdx].GetCommitmentsAlongPaths(keys)
```

Comment:
I imagine this line is the reason this function is taking an array of keys? IMO it'd be cleaner to take one key and do `[][]byte{key}` here and avoid `keys[0]` above